types/moo-indentation-lexer/index.d.ts

```diff
@@ -0,0 +1,55 @@
+import { Lexer, LexerState, Token } from "moo";
+
+export class IndentationLexer implements Lexer {
+    constructor({
+        lexer,
+        indentationType,
+        newlineType,
+        commentType,
+        indentName,
+        dedentName,
+        enclosingPunctuations,
+        separators,
+        state,
+        enclosures,
+        indentations,
+        queuedTokens,
+        queuedLines,
+        lastToken,
+    }: {
+        lexer: Lexer;
+        indentationType?: string;
+        newlineType?: string;
+        commentType?: string;
+        indentName?: string;
+        dedentName?: string;
+        enclosingPunctuations?: object;
+        separators?: string[];
+        state?: string;
+        enclosures?: Array<{ opening: string; indentationLevel: string }>;
+        indentations?: string[];
+        queuedTokens?: Token[];
+        queuedLines?: Token[][];
+        lastToken?: Token;
+    });
```

Comment:
It'd probably be worth moving this to a separate interface instead of having it inline here. It makes the code more readable and allows developers to more easily access the types.